domain/src/main/java/com/github/maleksandrowicz93/educational/institution/Faculty.java

```diff
@@ -0,0 +1,57 @@
+package com.github.maleksandrowicz93.educational.institution;
+
+import lombok.AccessLevel;
+import lombok.Builder;
+
+import java.util.Set;
+import java.util.stream.Collectors;
+
+@Builder(access = AccessLevel.PRIVATE)
+class Faculty implements Entity<FacultySnapshot> {
+
+    FacultyId id;
+    Name name;
+    Set<FieldOfStudy> fieldsOfStudy;
+    Set<Professor> professors;
+    Set<Course> courses;
+    Set<Student> students;
+
+    static Faculty from(FacultySnapshot snapshot) {
+        return builder()
+                .id(snapshot.id())
+                .name(snapshot.name())
+                .fieldsOfStudy(snapshot.fieldsOfStudy().stream()
+                        .map(FieldOfStudy::from)
+                        .collect(Collectors.toSet()))
+                .professors(snapshot.professors().stream()
+                        .map(Professor::from)
+                        .collect(Collectors.toSet()))
+                .courses(snapshot.courses().stream()
+                        .map(Course::from)
+                        .collect(Collectors.toSet()))
+                .students(snapshot.students().stream()
+                        .map(Student::from)
+                        .collect(Collectors.toSet()))
```

Comment:
statyczne importyyy

Reply:
Statyczne importy są niebezpieczne. Można łatwo tą metodę przysłonić / zhackować. Nie lubię tego stosować i boję się, że redukcja kodu przez statyczne importy to trochę sztuka dla sztuki. Przykład:
```java
@Getter
@Builder
@FieldDefaults(makeFinal = true)
public class Text {

    @Builder.Default
    String text = "yo";
    @Singular
    List<String> comments;
    @Singular
    Map<String, Integer> reviews;

    public boolean isValid() {
        return isNotBlank(text);
    }

    public boolean isNotBlank(String s) {
        return false;
    }
}
```

Reply:
jak wszystko w programowaniu - trzeba stosować z głową i świadomie

jeśli masz w klasie o odpowiedzialności innej niż collector metodę `toList()`/`toSet()` zwracającą `Collector`, to nie ze statycznymi importami jest problem

Reply:
Trochę dmuchałem na zimne, ale w sumie fakt - nie powinien się chyba bronić przed tym, że ktoś kiedyś może zrobić coś kiepskiego w kodzie. Poczytałem trochę i ogólnie lepiej jest korzystać z tego oszczędnie i rozsądnie, ale w niektórych przypadkach zdecydowanie może poprawić to czytelność klasy. Ale też trzeba uważać, bo może czasami czytelność popsuć :)

Reply:
Tutaj zdecydowanie '''javacollect(toSet())''' jest eleganckie, kontekst jest oczywisty, metoda używana bardzo często, więc ostatecznie z czystym sumieniem mogę to poprawić.